src/console.rs

```diff
@@ -73,10 +82,56 @@ impl Console {
 
         Ok(())
     }
+
+    fn set_global_console(self) {
+        unsafe {
+            CONSOLE = Some(self);
+        }
+    }
 }
 
 impl fmt::Write for Console {
     fn write_str(&mut self, string: &str) -> Result<(), fmt::Error> {
         self.write(string).map_err(|_| fmt::Error)
     }
 }
+
+#[macro_export]
+macro_rules! println {
+    () => ({
+        // Allow an empty println!() to print the location when hit
+        println!("")
+    });
+    ($msg:expr) => ({
+        if let Some(console) = $crate::console::get_global_console() {
+            use core::fmt::Write;
+            writeln!(console, $msg).unwrap()
```

Comment:
Wouldn't this generate a lot of panicking code?

Reply:
The `unwrap()`? That's what a lot of the current callers to `writeln!()` are already doing anyway. I also don't think it can really fail, so I don't this this panicing very often.

Reply:
Looking at the size diff github action (if I am reading it correctly), this PR seems to increase code size of most apps by 200-300 bytes and increase RAM use by ~60 bytes, which seems reasonable

Reply:
This is more or less what I expected. It doesn't matter whether the code will fail or not. The panicking code needs to be included in the binary and it contains a lot of meta information about each possible panic source.

There is an interesting article about how unwraps/panics have a bad impact not necessarily on code size but on runtime speed: http://likebike.com/posts/How_To_Write_Fast_Rust_Code.html

While it is okay to `unwrap` in the examples, this should never be done in the library code. That way, the library user can no longer collect multiple `Results` into a single `unwrap` via `?`.

I would suggest to avoid the `unwrap` call, either by writing `let _ = ...` or by calling the panic handler directly.

Reply:
Ok, I have stopped calling `unwrap()`. We now just ignore the errors (which I don't think can actually occur) with `let _ = `.